pymc/distributions/mixture.py

```diff
@@ -398,6 +404,22 @@ def marginal_mixture_logprob(op, values, rng, weights, *components, **kwargs):
     return mix_logp
 
 
+@_get_moment.register(MarginalMixtureRV)
+def get_moment_marginal_mixture(op, rv, rng, weights, *components):
+    ndim_supp = components[0].owner.op.ndim_supp
+
+    if len(components) == 1:
+        weights = at.shape_padright(weights, ndim_supp)
```

Comment:
I think this broadcasting is needed regardless of single or multiple components. It should be obvious with multivariate distributions, but with univariate (ndim_supp =0) distributions you wouldn't notice any problems

Reply:
Right! I ran into an error with a multivariate moment test,  I took out this line outside the `if` statement